### DISCUSSION TITLE
finding all duplicate data present in Milvus DB

### DISCUSSION THREAD

**Discussion by anic1618** [milvus-io/milvus#30929] [Q&A and General discussion]
How to find all duplicate records present in collection efficiently. 
one simple way is N^2 searches. can we do in N(N+1)/2 searches ? Any other efficient way?

**Comment by yhmo** [comment]
duplicate primary key, or duplicate vector?

**Reply by xiaofan-luan** [reply]
what is  N^2 searches? any ideas?

**Reply by anic1618** [reply]
Thanks.
Here duplicate records means duplicate vector. 
Simpler way will be Single vector search one by one/batches which will need  ~O(N^2) operations. What will be the  fasted way to do.

**Reply by yhmo** [reply]
If you have a copy of the vectors stored outside Milvus, then you do not need to fetch vectors from Milvus which will save a lot of time.

Batch search is more efficient than one-by-one.

**Reply by anic1618** [reply]
Thanks. 
But can there be more effective way? My task is finding all similar vector (By measure : IP in range (0.5 to 1)) in collection.

**Reply by yhmo** [reply]
Let's say you have 10000 vectors to find duplication. In Milvus collection, there are one million vectors.
We can:
```
for i in range(100):
    result = collection.search(data = [100 vectors])
```

Or:
```
for i in range(1000):
    result = collection.search(data = [10 vectors]
```

I think the 100x100 is a bit faster.

**Reply by anic1618** [reply]
Thanks.
I think I am not able to describe my problem.
Task: There are 160 M (let say n) embedding. We want to find similar embedding based on cosine distance. This required FLAT indexing. Doining search after taking all embedding batch by batch still requires n*n comparison (no sure) while we know know that i can be done in n*(n+1)/2 comparison by others means. I want to know what should be the most efficient apporach.
